modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/raft/RaftStorageManager.java

```diff
@@ -108,10 +110,65 @@ void removeLogicalTopologyNodes(Set<ClusterNode> nodes) {
     }
 
     private static byte[] logicalTopologyKey(ClusterNode node) {
-        byte[] nodeIdBytes = node.id().getBytes(StandardCharsets.UTF_8);
+        return prefixedKey(LOGICAL_TOPOLOGY_PREFIX, node.id());
+    }
+
+    /**
+     * Retrieves the validation token for a given node.
+     *
+     * @return Validation token or {@code null} if it does not exist.
+     */
+    @Nullable
+    UUID getValidationToken(ClusterNode node) {
+        return get(validationTokenKey(node.id()), UUID.class);
+    }
+
+    /**
+     * Saves the validation token for a given node.
+     */
+    void putValidationToken(ClusterNode node, UUID token) {
+        storage.put(validationTokenKey(node.id()), ByteUtils.toBytes(token));
+    }
+
+    /**
+     * Removes the validation token for a given node.
+     */
+    void removeValidationToken(String nodeId) {
+        storage.remove(validationTokenKey(nodeId));
+    }
+
+    private static byte[] validationTokenKey(String nodeId) {
+        return prefixedKey(VALIDATION_TOKEN_PREFIX, nodeId);
+    }
+
+    /**
+     * Returns a collection of node IDs that passed the validation (i.e. node IDs which have a corresponding validation token).
+     */
+    Collection<String> getValidatedNodeIds() {
```

Comment:
Naming is a bit confusing, cause such nodes will only be available during node join process. Could you please extent the javadoc?